### PR TITLE
Migrate TableTestBase related classes to JUnit5

### PR DESCRIPTION
**Pull request by tomtongue**
Migrate the following test classes to delete `TableTestBase`  for https://github.com/apache/iceberg/issues/9085.

This PR is for the migration of `TableTestBase` related classes in https://github.com/apache/iceberg/pull/10063.

## Current Progress

Core:
- [x] (skipped) `TestContentFileParser`
- [x] (skipped) `TestFileScanTaskParser` 
- [x] `util/TestTableScanUtil`

- [x] `WriterTestBase`
  - [x] `TestFileWriterFactory`
    - [x] `TestGenericFileWriterFactory`
    - [x] `TestSparkFileWriterFactory` for the versions: v3.3, v3.4, v3.5
    - [x] `TestFlinkFileWriterFactory` for the versions: v1.16, v.1.17, v1.18
  - [x] `TestPartitioningWriters`
    - [x] `TestSparkPartitioningWriters` for the versions: v3.3, v3.4, v3.5
    - [x] `TestFlinkPartitioningWriters` for the versions: v1.16, v.1.17, v1.18
  - [x] `TestPositionDeltaWriters`
    - [x] `TestSparkPositionDeltaWriters` for the versions: v3.3, v3.4, v3.5
    - [x] `TestFlinkPositionDeltaWriters` for the versions: v1.16, v.1.17, v1.18
  - [x] `TestRollingFileWriters`
    - [x] `TestSparkRollingFileWriters` for the versions: v3.3, v3.4, v3.5
    - [x] `TestFlinkRollingFileWriters` for the versions: v1.16, v.1.17, v1.18

`iceberg-flink` for the versions: v1.16, v.1.17, v1.18
- [x] `TestStreamingReaderOperator` 
- [x] `TestStreamingMonitorFunction`
- [x] `TestIcebergFilesCommitter`
- [x] `TestDeltaTaskWriter`

### CODE REVIEW

File: data/src/test/java/org/apache/iceberg/io/TestRollingFileWriters.java [hunk 1]
```diff
@@ -169,9 +170,9 @@ public void testRollingEqualityDeleteWriterSplitDeletes() throws IOException {
     writer.close();
 
     DeleteWriteResult result = writer.result();
-    Assert.assertEquals(4, result.deleteFiles().size());
-    Assert.assertEquals(0, result.referencedDataFiles().size());
-    Assert.assertFalse(result.referencesDataFiles());
+    assertThat(writer.result().deleteFiles()).hasSize(4);
```
Comment (nastra) [inline data/src/test/java/org/apache/iceberg/io/TestRollingFileWriters.java:173]: you may want to remove the `result` variable if you're using it directly here

Comment (tomtongue) [reply data/src/test/java/org/apache/iceberg/io/TestRollingFileWriters.java:173]: Oh sorry, I will update this part.

**Review by nastra — APPROVED** [review]
the changes you have so far LGTM, is this PR still WIP or ready?

**Comment by tomtongue** [comment]
> the changes you have so far LGTM, is this PR still WIP or ready?

Thanks for the review. This PR is ready, and all the classes are already migrated. Remove "WIP".

**Pull request merged**